src/main/java/com/minecolonies/rankup/util/AccountingUtils.java

```diff
@@ -0,0 +1,524 @@
+
+                if (conn != null)
+                {
+                    Statement stmt = conn.createStatement();
+
+                    stmt.execute("INSERT INTO " + table_id
+                                   + "(" + UUID_COLUMN + ", " + PLAYER_NAME_COLUMN + ", " + JOIN_DATE_COLUMN + ", " + LAST_JOIN_COLUMN + ", " + TIME_PLAYED_COLUMN + ") "
+                                   + "VALUES"
+                                   + "('" + uuid + "',"
+                                   + " '" + user.getName() + "',"
+                                   + " " + toDate(CommonUtils.dateNow()) + ","
+                                   + " " + toDate(CommonUtils.dateNow()) + ","
+                                   + " 0)");
+                }
+            }
+            else
+            {
+                final AccountConfigData accConfig = getAccountConfig();
+                accConfig.playerData.put(user.getUniqueId(), new AccountConfigData.PlayerConfig());
+                accConfig.playerData.get(user.getUniqueId()).timePlayed = 0;
+                accConfig.playerData.get(user.getUniqueId()).lastVisit = CommonUtils.dateNow();
+                accConfig.playerData.get(user.getUniqueId()).joinDate = CommonUtils.dateNow();
+                accConfig.playerData.get(user.getUniqueId()).playerName = user.getName();
+                accConfig.save();
+            }
+        }
+        catch (SQLException | NoModuleException e)
+        {
+            if (e instanceof SQLException)
+            {
+                plugin.getLogger().info(((SQLException) e).getSQLState());
+            }
+
+            e.printStackTrace();
+        }
+    }
+
+    public void updatePlayer(final UUID uuid)
+    {
+        final User user = Sponge.getServiceManager().provideUnchecked(UserStorageService.class).get(uuid).orElse(null);
+
+        if (user == null || !doesPlayerExist(uuid))
+        {
+            return;
+        }
+
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                if (!user.getName().equals(getPlayerName(uuid)))
+                {
+                    updatePlayerName(uuid, user.getName());
+                }
+
+                updatePlayerLastDate(uuid, CommonUtils.dateNow());
+            }
+            else
+            {
+                final AccountConfigData accConfig = getAccountConfig();
+                accConfig.playerData.put(user.getUniqueId(), new AccountConfigData.PlayerConfig());
+                accConfig.playerData.get(user.getUniqueId()).timePlayed = 0;
+                accConfig.playerData.get(user.getUniqueId()).lastVisit = CommonUtils.dateNow();
+                accConfig.playerData.get(user.getUniqueId()).joinDate = CommonUtils.dateNow();
+                accConfig.playerData.get(user.getUniqueId()).playerName = user.getName();
+                accConfig.save();
+            }
+        }
+        catch (NoModuleException e)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1166)